### PR TITLE
Fetch components without openlcs_scan_url

### PR DESCRIPTION
**Pull request by qduanmu**

### CODE REVIEW

File: openlcs/libs/corgi.py [hunk 1]
```diff
@@ -591,6 +591,8 @@ def should_yield_data(component, result):
             # subscription purls obtained from previous sync
             subscribed_purls = subscription.get("component_purls", [])
 
+            # Only collect components without openlcs_scan_url
+            query_params.update({'missing_scan_url': True})
```
Comment (hanchuntao) [inline openlcs/libs/corgi.py:595]: where use this `missing_scan_url ` ?

Comment (qduanmu) [reply openlcs/libs/corgi.py:595]: This is the new filter added by Corgi.

Comment (hanchuntao) [reply openlcs/libs/corgi.py:595]:
```
    missing_scan_url = EmptyStringFilter(
        field_name="openlcs_scan_url",
        label="Show only unscanned components (where OpenLCS scan URL is empty)",
    )
```
Thanks for your explanation.

**Comment by hanchuntao** [comment]
LGTM

**Comment by RichardXuan** [comment]
I think we should also filter nested type component's sub component like OCI and RPMMOD when get data from `provides ` field.

**Comment by qduanmu** [comment]
That's why it was not merged as I highlighted in previous meeting, it doesn't make sense for oci and rpmmod as there is no scan url provided.

**Comment by RichardXuan** [comment]
Ok, got it

**Comment by qduanmu** [comment]
There is a data problem in populating subscription purls, bug filed, will add a check after that issue fixed, close this.

**Pull request closed without merging**